request.go

```diff
@@ -389,6 +419,78 @@ func (conn *Connection) ExecuteAsync(expr string, args interface{}) *Future {
 	return conn.DoAsync(req)
 }
 
+// PrepareAsync sends a sql statement to prepare and returns PreparedStatement object.
+//
+// Since 1.7.0
+func (conn *Connection) PrepareAsync(expr string) *PreparedStatement {
+	req := newPrepareRequest(expr)
+	fut := conn.DoAsync(req)
+	stmt := newPreparedStatement(fut, conn)
+	return stmt
+}
```

Comment:
I don't like the idea of using futures for a work with prepared statements. The Future object looks too raw for it, we can catch some situations when it is too complicated or not safe to use it. We can't prevent the calling GetTyped with a future for example. PrepareAsync returns the object immediately like Future when we call DoAsync.

Reply:
> The Future object looks too raw for it, we can catch some situations when it is too complicated or not safe to use it. We can't prevent the calling GetTyped with a future for example.

I think you're overstating the problem. You can add a simple check into Future.GetTyped(). Three basic tests will be enough to ensure that all ok (and probably the same number of tests for not ok cases):

1. Connection.PrepareAsync().Get()
2. Connection.PrepareAsync().GetTyped()
3. Connection.PrepareAsync().GetIterator()

Reply:
Please look at this commit 8a538fa528806ee305c8235891baff0120130189. What do you thin about using it so?

Reply:
> Please look at this commit [8a538fa](https://github.com/tarantool/go-tarantool/commit/8a538fa528806ee305c8235891baff0120130189). What do you think about using it so?

I think it looks great from the `Connection` point of view:

```Go
// From that point of view it will be better to use PreparedStatementID (an user will be
// able to create a prepared statement object created in an external place)
stmt := NewPreparedStatement(GetPreparedStatementtId(resp))
conn.DoAsync(NewPreparedExecuteRequest(stmt))
conn.DoAsync(NewUnprepareRequest(stmt))
```

But we'll have to do something with `ConnectionPool`.